library/network/src/modules/NetworkInterfaces.rb

```diff
@@ -799,48 +806,42 @@ def FilterDevices(devregex)
     end
 
     def FilterNOT(devices, devregex)
-      devices = deep_copy(devices)
       return {} if devices.nil? || devregex.nil? || devregex == ""
+      devices = deep_copy(devices)
 
-      regex = Ops.add(
-        Ops.add("^(", Ops.get(@DeviceRegex, devregex, devregex)),
-        ")[0-9]*$"
-      )
-      Builtins.y2debug("regex=%1", regex)
-      devices = Builtins.filter(devices) do |file, _devmap|
-        Builtins.regexpmatch(file, regex) != true
-      end
-      Builtins.y2debug("devices=%1", devices)
-      deep_copy(devices)
+      regex = "^(#{@DeviceRegex[devregex] || devregex})[0-9]*$"
+
+      log.debug("regex=#{regex}")
+      devices.select! { |f, _d| f !~ regex }
```

Comment:
The differences are in the return and in the operator, or we could use same operator but reject.. but in that case we need other parameter. Is it worth it?